### PR TITLE
leaflet: Calc - Accept Keyboard input on mobile without tapping on cell

### PR DESCRIPTION
**Pull request by Ezinnem**
…he keyboard is availble

Signed-off-by: Ezinne Nnamani <nnamani.ezinne@collabora.com>
Change-Id: I2ea3e9248eacaf3923ef1633ac5aa8838241655e
Signed-off-by: Ezinne Nnamani <nnamani.ezinne@collabora.com>

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: loleaflet/src/control/Permission.js [hunk 1]
```diff
@@ -81,8 +81,10 @@ L.Map.include({
 			this.fire('editorgotfocus');
 			// In the iOS/android app, just clicking the mobile-edit-button is
 			// not reason enough to pop up the on-screen keyboard.
-			if (!(window.ThisIsTheiOSApp || window.ThisIsTheAndroidApp))
-				this.focus();
+			if (!(window.ThisIsTheiOSApp || window.ThisIsTheAndroidApp)) {
+				this.focus(true);
+				this._docLayer.postKeyboardEvent('input', 0, 769); //F2
+			}
```
Comment (Ashod) [inline loleaflet/src/control/Permission.js:87]: This is great for Calc, but we probably don't want (or need) to send F2 for Writer and Impress.

**Review by Ashod — CHANGES_REQUESTED** [review]
Thanks Ezinne for the patch! Looks good (as discussed) but I think we need to isolate it (see comment in code).

Another suggestion is to make the commit message much more detailed and descriptive, so others can understand 1) what the issue was, 2) how you fixed it, 3) why this approach and not another, 4) optional: any other comments/caveats for the future (for example a todo to improve something). Also, making the title message short with a longer body is best for readability. Notice how we often prefix the title message with the domain/area so that others would know what this change is about without looking at the patch itself.

Here is a suggestion (feel free to modify):

leaflet: calc: allow keyboard input on mobile without tapping

On mobile Calc required tapping a cell to start editing.
To allow for seamless editing right after clicking the
edit button in mobile view, we need to send Core F2,
which enters edit mode and we get the cursor position.

The reason we need to send F2 explicitly, even though
this isn't needed on desktop, is because in mobile we
start in read-only mode, and therefore we don't have
the cursor position. So we need to initiate editing when
the user clicks the edit button and enables editing again.

**Comment by thebearon** [comment]
Let me point out that the original request was about staying in cell edit mode when you tap/press enter after entering content in a cell, and not about switching to editing the file after the initial read-only view.

File: loleaflet/src/control/Permission.js [hunk 2]
```diff
@@ -81,8 +81,12 @@ L.Map.include({
 			this.fire('editorgotfocus');
 			// In the iOS/android app, just clicking the mobile-edit-button is
 			// not reason enough to pop up the on-screen keyboard.
-			if (!(window.ThisIsTheiOSApp || window.ThisIsTheAndroidApp))
-				this.focus();
+			if (!(window.ThisIsTheiOSApp || window.ThisIsTheAndroidApp)) {
+				if (this._docLayer._docType === 'spreadsheet') {
```
Comment (Ashod) [inline loleaflet/src/control/Permission.js:85]: Don't we want to take focus for all document types? F2 is meaningful only for Calc, but the focus is for all.

Also, I think there is .isCalc() helper function somewhere. Can you check if we can use it here please? Comparing with strings is error prone, so best to minimize/avoid them.

**Comment by kendy** [comment]
Agreed with @Ezinnem and @Ashod to close this.  Thank you for the patch either way! :-)

**Pull request closed without merging**